rotkehlchen/api/v1/fields.py

```diff
@@ -66,6 +67,51 @@
 log = RotkehlchenLogsAdapter(logger)
 
 
+class IncludeExcludeListField(fields.Field):
+    """ A field that accepts an object like the following and deserializes it to the proper types.
+    {
+        "values": [val1, val2, val3],
+        "behaviour": "exclude"
+    }
+    Eventually, this object will be used to create a filter query to the db, to return all the db
+    table entries where this field's value is not in [val1, val2, val3]. The behaviour key
+    can be either "include" or "exclude" and it defaults to "include" if not present.
+    """
+    def __init__(
+            self,
+            values_field: 'SerializableEnumField',
+            *args: Any,
+            **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.values_field = values_field
+
+    def _deserialize(
+            self,
+            value: dict[str, Union[list, str]],
+            attr: Optional[str],
+            data: Any,
+            **kwargs: Any,
+    ) -> IncludeExcludeFilterData:
+
+        if 'behaviour' not in value:
+            # If for any reason the behaviour key is not given, we use the default behaviour.
+            value['behaviour'] = 'include'
+
+        values = value['values']
+        behaviour = value['behaviour']
+
+        if behaviour not in ['include', 'exclude']:
+            raise ValidationError("Behaviour must be either 'include' or 'exclude'.")
+
+        deserialized_values = DelimitedOrNormalList(self.values_field).deserialize(values)
+        deserialized_behaviour = cast(Literal['IN', 'NOT IN'], 'IN' if behaviour == 'include' else 'NOT IN')  # noqa: E501
```

Comment:
cast is one way but annoying and does not help or tell us anything. In this case I would accept a type:ignore[XXX] where XXX is the name of the violation, which I assume is it does not see that it's Literal for some reason? But why?

```suggestion
        deserialized_behaviour = 'IN' if behaviour == 'include' else 'NOT IN'  # type: ignore[XXX]
```